src/editor.rs

```diff
@@ -124,12 +124,16 @@ impl StatusMessage {
 
 /// Pretty-format a size in bytes.
 fn format_size(n: u64) -> String {
-    #![allow(clippy::cast_precision_loss)]
-    let quo_rem = successors(Some((n, 0)), |(q, _)| Some((q / 1024, q % 1024)).filter(|u| u.0 > 0));
-    // unwrap(): quo_rem is never empty (since `successors` has an initial value), so _.last()
-    // cannot be None
-    let ((q, r), prefix) = quo_rem.zip(&["", "k", "M", "G", "T", "P", "E", "Z"]).last().unwrap();
-    format!("{:.2$}{}B", q as f32 + r as f32 / 1024., prefix, p = if *prefix == "" { 0 } else { 2 })
+    if n < 1024 {
+        return format!("{}B", n);
+    }
+    // i is the largest value such that 1024 ^ i < n
+    // To find i we compute the smallest b such that n <= 1024 ^ b and subtract -1 from b
+    let i = ((64 - n.leading_zeros()) as f64 / 10.0).ceil() as usize - 1;
```

Comment:
I think this is equivalent and doesn't require casting to `f64`:

```suggestion
    let i = ((64 - n.leading_zeros() + 9) / 10) as usize - 1;
```

Reply:
Neat.